overlord/snapstate/backend/link.go

```diff
@@ -255,14 +261,19 @@ func removeGeneratedSnapdWrappers(s *snap.Info, firstInstall bool, meter progres
 // symlinks. The firstInstallUndo is true when undoing the first installation of
 // the snap.
 func (b Backend) UnlinkSnap(info *snap.Info, linkCtx LinkContext, meter progress.Meter) error {
+	// inhibit startup of new programs
+	// TODO: we probably need a dedicated field to provide the hint.
+	inhibitHint := runinhibit.HintInhibitedForRefresh
+	err0 := runinhibit.LockWithHint(info.InstanceName(), inhibitHint)
```

Comment:
shouldn't the unit tests for UnlinkSnap and LinkSnap check for the added behavior?

Reply:
Done